tests/core/utilities/test_method_formatters.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+
+from web3._utils.method_formatters import (
+    get_error_formatters,
+    get_revert_reason,
+)
+from web3._utils.rpc_abi import (
+    RPC,
+)
+from web3.exceptions import (
+    SolidityError,
+)
+from web3.types import (
+    RPCResponse,
+)
+
+REVERT_WITH_MSG = RPCResponse({
+    'jsonrpc': '2.0',
+    'error': {
+        'code': -32015,
+        'message': 'VM execution error.',
+        'data': (
+            'Reverted '
+            '0x08c379a'
+            '00000000000000000000000000000000000000000000000000000000000000020'
+            '0000000000000000000000000000000000000000000000000000000000000016'
+            '6e6f7420616c6c6f77656420746f206d6f6e69746f7200000000000000000000'
+        ),
+    },
+    'id': 2987,
+})
+
+REVERT_WITHOUT_MSG = RPCResponse({
+    'jsonrpc': '2.0',
+    'error': {
+        'code': -32015,
+        'message': 'VM execution error.',
+        'data': 'Reverted 0x',
+    },
+    'id': 2987,
+})
+
+OTHER_ERROR = RPCResponse({
+    "jsonrpc": "2.0",
+    "error": {
+        "code": -32601,
+        "message": "Method not found",
+    },
+    "id": 1,
+})
+
+
+def test_get_revert_reason() -> None:
```

Comment:
This test seems like it's worth [parameterizing](https://docs.pytest.org/en/stable/parametrize.html). Right now - for each successful test run we're testing 3 different case. Each test run should assert/disprove 1 case. 

Also the next test seems like it should be parametrized as well for the two cases that test if a `SolidityError` is raised, and then the `OTHER_ERROR` case split into its own test.